spec/services/cache_service_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper'
+
+describe CacheService do
+  let(:rails_cache) { Rails.cache }
+
+  describe "#cache" do
+    before do
+      rails_cache.stub(:fetch)
+    end
+
+    it "provides a wrapper for basic #fetch calls to Rails.cache" do
+      CacheService.cache("test-cache-key", expires_in: 10.seconds) do
+        "TEST"
+      end
+
+      expect(rails_cache).to have_received(:fetch).with("test-cache-key", expires_in: 10.seconds)
+    end
+  end
+
+  describe "#cached_data_by_class" do
+    let(:timestamp) { Time.now.to_i }
+
+    before do
+      rails_cache.stub(:fetch)
+      CacheService.stub(:latest_timestamp_by_class) { timestamp }
+    end
+
+    it "caches data by timestamp for last record of that class" do
+      CacheService.cached_data_by_class("test-cache-key", Enterprise) do
+        "TEST"
+      end
+
+      expect(CacheService).to have_received(:latest_timestamp_by_class).with(Enterprise)
+      expect(rails_cache).to have_received(:fetch).with("test-cache-key-Enterprise-#{timestamp}")
```

Comment:
This spec validates the internals of the implementation, not the behavior.
We shouldnt stub internal methods of the class under test. Only it's external limits.
I'd expect Enterprise.maximum to be mocked here and the expectation is that the rails cache is called with that value.

```ruby
  before do
      expect(rails_cache.stub(:fetch)
      allow(Enterprise).to receive(:maximum).with(:updated_at).and_return(timestamp)
    end

    it "caches data by timestamp for last record of that class" do
      CacheService.cached_data_by_class("test-cache-key", Enterprise) do
        "TEST"
      end

      expect(rails_cache).to have_received(:fetch).with("test-cache-key-Enterprise-#{timestamp}")
```